Xamarin.Essentials/DeviceInfo/DeviceInfo.macos.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Foundation;
+using ObjCRuntime;
+
+namespace Xamarin.Essentials
+{
+    public static partial class DeviceInfo
+    {
+        [DllImport(Constants.SystemConfigurationLibrary)]
+        static extern IntPtr SCDynamicStoreCopyComputerName(IntPtr store, IntPtr encoding);
+
+        [DllImport(Constants.CoreFoundationLibrary)]
+        static extern void CFRelease(IntPtr cf);
+
+        static string GetModel() =>
+            IOKit.GetPlatformExpertPropertyValue<NSData>("model")?.ToString() ?? string.Empty;
+
+        static string GetManufacturer() => "Apple";
+
+        static string GetDeviceName()
+        {
+            var computerNameHandle = SCDynamicStoreCopyComputerName(IntPtr.Zero, IntPtr.Zero);
+
+            if (computerNameHandle == IntPtr.Zero)
+                return null;
+
+            var computerName = NSString.FromHandle(computerNameHandle).ToString();
+
+            try
+            {
+                CFRelease(computerNameHandle);
+            }
+            catch
+            {
+            }
+
+            return computerName;
```

Comment:
This looks strange, I'm assuming you're adding a try/catch around CFRelease to catch any exceptions; the problem is if that CFRelease goes wrong, you'll crash, and no exception handling will help.

I think this looks more natural:

```suggestion
            try
            {
                return NSString.FromHandle(computerNameHandle);
            }
            finally
            {
                CFRelease(computerNameHandle);
            }
```